ui/pages/settings/experimental-tab/experimental-tab.component.js

```diff
@@ -118,6 +120,36 @@ export default class ExperimentalTab extends PureComponent {
     }
   }
 
+  renderTogglePetnames() {
+    const { t } = this.context;
+    const { petnamesEnabled, setPetnamesEnabled } = this.props;
+
+    return (
+      <Box
```

Comment:
When I added the external source toggle previously, I noticed the rendering of each toggle is almost identical with each one adding a fair amount of duplicate HTML.

Not needed in this PR and not your fault, but I'm surprised there isn't a nested `PreferenceToggle` component to remove all this duplication.